### PR TITLE
resurrect and refactor lockdrop vault test #NTRN-462

### PR DESCRIPTION
**Pull request by foxpy**
- I have deleted `lockdrop.test.ts` since `tge.auction.ts` already implements almost everything. Basic tests from `lockdrop.test.ts` have been moved to `tge.auction.ts`.
- `tge.auction.ts` has been heavily refactored to simplify TGE deployment.
- I have deleted a TODO which has already been implemented in another test suite.
- Since `tge.auction.ts` tests vaults heavily anyway, I have added some config tests for them there.

[[Recent test run](https://github.com/neutron-org/neutron-tests/actions/runs/5119086199)]

### CODE REVIEW

**Comment by foxpy** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/5015273643

**Comment by foxpy** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/5033177551

**Comment by foxpy** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/5045278298

**Comment by foxpy** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/5078887877

File: src/testcases/run_in_band/tge.auction.test.ts [hunk 1]
```diff
@@ -2520,6 +2023,352 @@ describe('Neutron / TGE / Auction', () => {
...
+          await neutronChain.queryDenomBalance(
+            tgeWallets[
+              'airdropAuctionLockdropVesting'
+            ].wallet.address.toString(),
+            NEUTRON_DENOM,
+          );
+        const expectedLockdropReward = Number(
+          AirdropAuctionLockdropVestingUserInfo.total_ntrn_rewards,
+        );
+        const airdropSize = 100;
```
Comment (sotnikov-s) [inline src/testcases/run_in_band/tge.auction.test.ts:2037]: I'd really appreciate it if we had such numbers as constants defined somewhere and used across the tests

File: src/testcases/run_in_band/tge.auction.test.ts [hunk 2]
```diff
@@ -2366,15 +1822,42 @@ describe('Neutron / TGE / Auction', () => {
...
+              Number(airdropCNTRN.balance),
+          );
+        });
       });
     });
     describe('lockdrop', () => {
       let balanceBeforeLockdrop: number;
       let balanceBeforeAirdopLockdrop: number;
+      let balanceBeforeAirdropAuctionLockdropVesting: number;
+      let AirdropAuctionLockdropVestingUserInfo: LockDropInfoResponse;
```
Comment (sotnikov-s) [inline src/testcases/run_in_band/tge.auction.test.ts:1860]: please lowercase the first letter

File: src/helpers/tge.ts [hunk 3]
```diff
@@ -31,3 +40,1137 @@ export class Airdrop {
...
+ * tge.times.airdropStart = getTimestamp(0);
+ * tge.times.airdropVestingStart = getTimestamp(300);
+ * await tge.deployPreAuction();
+ * tge.times.auctionInit = getTimestamp(30);
+ * await tge.deployAuction();
+ * tge.times.lockdropInit = getTimestamp(10);
+ * await tge.deployLockdrop();
+ * await tge.deployLockdropVault();
+ *
+ */
```
[the remark this answers is not in the document]
Comment (foxpy) [reply src/helpers/tge.ts:69]: these are not commented lines, that's documentation…

File: src/helpers/tge.ts [hunk 4]
```diff
@@ -31,3 +40,1137 @@ export class Airdrop {
...
+    this.contracts.oracleUsdc = await instantiateAstroportOracle(
+      this.instantiator,
+      this.codeIds.ORACLE_HISTORY,
+      this.contracts.astroFactory,
+      1,
+      this.tokenInfoManager.wallet.address.toString(),
+      'astroport_oracle_usdc',
+    );
+
+    for (const [contract, denom1] of [
```
[the remark this answers is not in the document]
Comment (foxpy) [reply src/helpers/tge.ts]: made it more clear to understand, do you like it now?

File: src/testcases/run_in_band/tge.auction.test.ts [hunk 5]
```diff
@@ -281,583 +244,63 @@ describe('Neutron / TGE / Auction', () => {
...
-          vesting_migration_pack_size: 4,
-          vesting_lp_duration: times.auctionVestingLpDuration,
-        }),
-        'auction',
-      );
-      contractAddresses.TGE_AUCTION = res[0]._contract_address;
+    it('should deploy auction', async () => {
+      tge.times.auctionInit = getTimestamp(80);
+      tge.times.auctionDepositWindow = 60;
+      tge.times.auctionWithdrawalWindow = 60;
```
[the remark this answers is not in the document]
Comment (foxpy) [reply src/testcases/run_in_band/tge.auction.test.ts]: I have removed this duplicate logic

File: src/testcases/run_in_band/tge.auction.test.ts (hunk 1 above)
Comment (foxpy) [reply src/testcases/run_in_band/tge.auction.test.ts:2025]: @swelf19 that's your recent addition, could you please resolve this thread?

**Comment by foxpy** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/5112539129

**Review by oopcode — DISMISSED** [review]

**Comment by foxpy** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/5119086199

**Review by oopcode — APPROVED** [review]

**Review by sotnikov-s — APPROVED** [review]

**Pull request merged**